ts/src/connections/tunnelRelayTunnelClient.ts

```diff
@@ -1,87 +1,73 @@
-
-    constructor() {
-        super();
+    constructor(trace?: Trace, managementClient?: TunnelManagementClient) {
+        super(trace, managementClient);
     }
 
-    public async connectClient(tunnel: Tunnel, endpoints: TunnelEndpoint[]): Promise<void> {
-        let tunnelEndpoints = endpoints.map((endpoint) => endpoint as TunnelRelayTunnelEndpoint);
+    /**
+     * Gets the tunnel relay URI.
+     */
+    public async getTunnelRelayUri(tunnel?: Tunnel): Promise<string> {
+        let tunnelEndpoints = this.endpoints!.map(
+            (endpoint) => endpoint as TunnelRelayTunnelEndpoint,
+        );
         let tunnelEndpoint;
         if (tunnelEndpoints && tunnelEndpoints.length === 1) {
             tunnelEndpoint = tunnelEndpoints[0];
         } else {
             throw new Error('The host is not currently accepting Tunnel relay connections.');
         }
 
-        let clientRelayUri = tunnelEndpoint.clientRelayUri;
-        if (!clientRelayUri) {
-            throw new Error('The tunnel client relay endpoint URI is missing.');
-        }
-
-        let accessToken = tunnel.accessTokens
-            ? tunnel.accessTokens[TunnelAccessScopes.Connect]
-            : undefined;
-
-        await this.connectClientToRelayServer(clientRelayUri, accessToken);
+        return tunnelEndpoint.clientRelayUri!;
     }
 
+    /**
+     * Connect to the tunnel session on the relay service using the given access token for authorization.
+     */
     protected async connectClientToRelayServer(
         clientRelayUri: string,
         accessToken?: string,
     ): Promise<void> {
-        this.trace(TraceLevel.Info, 0, `Connecting to client tunnel relay ${clientRelayUri}`);
-        this.trace(
-            TraceLevel.Verbose,
-            0,
-            `Sec-WebSocket-Protocol: ${TunnelRelayTunnelClient.webSocketSubProtocol}`,
-        );
-        if (accessToken) {
-            const token = TunnelAccessTokenProperties.tryParse(accessToken)?.toString() ?? 'token';
-            this.trace(TraceLevel.Verbose, 0, `Authorization: tunnel <${token}>`);
+        if (!clientRelayUri) {
+            throw new Error('Client relay URI must be a non-empty string');
         }
 
-        try {
-            let stream = await this.streamFactory.createRelayStream(
-                clientRelayUri,
-                TunnelRelayTunnelClient.webSocketSubProtocol,
-                accessToken,
-            );
-            try {
-                await this.startSshSession(stream);
-            } catch {
-                stream.dispose();
-                throw new Error();
-            }
-        } catch (ex) {
-            throw new Error('Failed to connect to tunnel relay. ' + ex);
+        this.relayUri = clientRelayUri;
+        this.accessToken = accessToken;
+        await this.connectTunnelSession();
+    }
+
+    /**
+     * Configures the tunnel session with the given stream.
+     */
+    public async configureSession(
```

Comment:
Should this be internal? In TS, internal methods on exported classes can be marked with `/* @internal */` to prevent them from being included in the `.d.ts` type definitions. See https://www.typescriptlang.org/tsconfig#stripInternal

Maybe some other methods should be marked that way also.